CMakeLists.txt

```diff
@@ -337,12 +341,6 @@ if (CODE_COVERAGE)
 
   if (CMAKE_C_COMPILER_ID MATCHES GNU)
     message(FATAL_ERROR "Code coverage is currently only supported by clang.")
-  else ()
-    if (CMAKE_C_COMPILER_VERSION VERSION_LESS 7 OR CMAKE_C_COMPILER_VERSION
```

Comment:
We should keep the warning message to avoid potential conflicting with other compiler versions. Just update the version.